blog/tests/views_test.py

```diff
@@ -0,0 +1,140 @@
+"""Test for views."""
+
+from datetime import datetime
+
+from django.contrib.auth.models import User
+from django.test import TestCase
+from django.urls import reverse
+
+from ..models import Post
+
+
+class PostListTest(TestCase):
+    """Tests for Post views."""
+
+    def setUp(self):
+        """Set up non-modifiable objects used by all test methods."""
+        bob = User.objects.create(username='bob')
+        self.post_first = Post.objects.create(
+            author=bob,
+            title='Title Bob',
+            text='Text bob',
+            published_date=datetime(day=8, month=11, year=2018),
+        )
+
+        john = User.objects.create(username='john')
+
+        self.post_second = Post.objects.create(
+            author=john, title='Title John', text='Text John'
+        )
+        self.post_third = Post.objects.create(
+            author=john, title='Title John 2', text='Text John 2'
+        )
+        self.post_third.publish()
+
+    def tearDown(self):
+        """Clean-up test data."""
+        del self.post_first
+        del self.post_second
+        del self.post_third
+
+    def test_post_list_correct_order(self):
+        """Test show post list on the page and test correct order."""
+        response = self.client.get(reverse('post_list'))
+        assert response.status_code == 200
+
+        self.assertTemplateUsed(
+            response, 'blog/post_list.html', 'blog/base.html'
+        )
+
+        assert len(response.context['posts']) == 2
+
+        first_title_on_the_page = response.context['posts'][0].title
```

Comment:
This is too verbose and manual comparison. You can just compare post lists, not individual attributes.

Here's a good example of how to check what's rendered in different points of time: https://github.com/kpi-web-guild/django-girls-blog-makkaronis4e/pull/27/files#diff-8e9b1eecb0153fa41f8d69ca4922dc61R35